packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-javascript-visitor.test.ts.snap

```diff
@@ -52,7 +52,7 @@ export declare class Post {
   };
   readonly id: string;
   readonly title: string;
-  readonly comments?: (Comment | null)[] | null;
+  readonly comments?: (LazyLoading extends Disabled ? (Comment | null)[] : AsyncCollection<Comment>) | null;
```

Comment:
Similarly to the above, I think we want to move the optionality into the type unions so that the collection is always present -- so that the developer doesn't have to drop `?`'s and/or `!`'s everywhere.

We had also decided that due to the nature of a has-many, we should always return an `AsyncCollection` for Lazy Loading, but that it may simply be empty. So, I think the backwards compatible version needs to look something like this:

```
  readonly comments: LazyLoading extends Disabled ? ((Comment | null)[] | undefined) : AsyncCollection<Comment>;
```